### PR TITLE
Add out of vocabulary error to ASR models

### PR DESCRIPTION
**Pull request by will-rice**
# What does this PR do?
This PR adds a check to Wav2Vec2 and Hubert models to ensure the labels do not contain values greater than the configured vocab size.  

Fixes # (issue)
https://github.com/huggingface/transformers/issues/12270

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@patrickvonplaten 

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/hubert/modeling_hubert.py [hunk 1]
```diff
@@ -1030,6 +1030,9 @@ def forward(
         loss = None
         if labels is not None:
 
+            if labels.max() > self.config.vocab_size:
```
Comment (patrickvonplaten) [inline src/transformers/models/hubert/modeling_hubert.py:1033]: Think it should actually be `>=` no? 

If a vocabulary has a vocab_size of 32, then the ids 0 - 31 are used I think. So having a 32 in the labels would fail IMO

**Review by patrickvonplaten — COMMENTED** [review]
Hey @will-rice,

The PR looks great - thanks a lot for adding the tests as well. I think we just need to change `>` to `>=`
